src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/IssueDocumentHandlerTest.java

```diff
@@ -147,4 +148,62 @@ public void testDocumentPayloadValues() {
         assertEquals(localDate, payload.getGeneratedDate());
         assertEquals("Barry Allen", payload.getIdamSurname());
     }
+
+    @Test
+    public void givenSetAsideState_thenReturnSetAsideDecisionNotice() {
+        SscsCaseData sscsCaseData = SscsCaseData.builder()
+            .ccdCaseId("1")
+            .postHearing(PostHearing.builder()
+                .setAside(SetAside.builder()
+                    .action(SetAsideActions.GRANT)
+                    .build())
+                .build())
+            .build();
+
+        String documentTypeLabel = new IssueDocumentHandler().getDocumentTypeLabel(sscsCaseData, DocumentType.DECISION_NOTICE, "adfdsf2");
+
+        assertEquals("Set Aside Decision Notice", documentTypeLabel);
+    }
+
+    @Test
+    public void givenSetAsideStateRefused_thenReturnSetAsideDecisionNotice() {
+        SscsCaseData sscsCaseData = SscsCaseData.builder()
+            .ccdCaseId("1")
+            .postHearing(PostHearing.builder()
+                .setAside(SetAside.builder()
+                    .action(SetAsideActions.REFUSE)
+                    .build())
+                .build())
+            .build();
+
+        String documentTypeLabel = new IssueDocumentHandler().getDocumentTypeLabel(sscsCaseData, DocumentType.DECISION_NOTICE, "adfdsf2");
+
+        assertEquals("Set Aside Decision Notice", documentTypeLabel);
+    }
+
+    @Test
+    public void givenSetAsideStateIsNull_thenReturnDraftDecisionNotice() {
+        SscsCaseData sscsCaseData = SscsCaseData.builder()
+            .ccdCaseId("1")
+            .postHearing(PostHearing.builder()
+                .setAside(SetAside.builder()
+                    .action(null)
+                    .build())
+               .build())
+            .build();
+
+        String documentTypeLabel = new IssueDocumentHandler().getDocumentTypeLabel(sscsCaseData, DocumentType.DECISION_NOTICE, "Draft Decision Notice");
+        assertEquals("Draft Decision Notice", documentTypeLabel);
```

Comment:
Please could you explain how these bottom two tests work and what they are testing. "Draft Decision Notice" is the expected result, but it appears to be set in the line above. I'm just unsure how the case data plays a role.

Reply:
1) The first test case will be used to test when the **PostHearing** is Null
2) Second test case is when the **setAside.action** is null
In both the case the Document type name should be **Draft Decision Notice**